### PR TITLE
Add ci

### PR DESCRIPTION
**Pull request by movsic**

### CODE REVIEW

File: .github/workflows/_all_ci.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      working-directory:
+        required: true
+        type: string
+        description: "From which folder this pipeline executes"
+  workflow_dispatch:
+    inputs:
+      working-directory:
+        required: true
+        type: choice
+        default: 'libs/langchain'
```
Comment (lkuligin) [inline .github/workflows/_all_ci.yml:16]: should we change the default?

File: .github/workflows/_dependencies.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+          # Check that the major part of pydantic version is as expected, if not
+          # raise an error
+          if [[ "$CURRENT_VERSION" != "$TEST_WITH_VERSION" ]]; then
+            echo "Error: expected pydantic version ${CURRENT_VERSION} to have been installed, but found: ${TEST_WITH_VERSION}"
+            exit 1
+          fi
+          echo "Found pydantic version ${CURRENT_VERSION}, as expected"
+      - name: Run pydantic compatibility tests
+        shell: bash
+        run: make test
```
Comment (lkuligin) [inline .github/workflows/_dependencies.yml:101]: do we need a Makefile for this?

File: .github/workflows/_lint.yml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        with:
+          path: |
+            ${{ env.WORKDIR }}/.mypy_cache
+          key: mypy-lint-${{ runner.os }}-${{ runner.arch }}-py${{ matrix.python-version }}-${{ inputs.working-directory }}-${{ hashFiles(format('{0}/poetry.lock', inputs.working-directory)) }}
+
+
+      - name: Analysing the code with our lint
+        working-directory: ${{ inputs.working-directory }}
+        run: |
+          make lint_package
```
Comment (lkuligin) [inline .github/workflows/_lint.yml:95]: do we need a Makefile for this?

File: .github/workflows/_dependencies.yml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        working-directory: ${{ inputs.working-directory }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version:
+          - "3.8"
+          - "3.9"
+          - "3.10"
+          - "3.11"
+    name: dependency checks ${{ matrix.python-version }}
```
Comment (lkuligin) [inline .github/workflows/_dependencies.yml:31]: do we really need this one?

File: .github/workflows/_all_ci.yml (hunk 1 above)
Comment (movsic) [reply .github/workflows/_all_ci.yml]: fixed

**Review by lkuligin — APPROVED** [review]

**Review by efriis — APPROVED** [review]

**Pull request merged**